src/app/(frontend)/[id]/page.tsx

```diff
@@ -0,0 +1,92 @@
+"use client"
+import { useParams } from 'next/navigation';
+import React, { useState, useEffect } from 'react';
+import { Product } from '@/lib/models/ProductModel';
+import Image from 'next/image';
+import { register } from 'swiper/element/bundle';
+
+register();
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'swiper-container': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & {
+        navigation?: string;
+        pagination?: string;
+        loop?: string;
+      }, HTMLElement>,
+      'swiper-slide': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>,
+    }
+  }
+}
+
+const ProductDetailPage: React.FC = () => {
+  const [product, setProduct] = useState<Product | null>(null)
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
```

Comment:
I´d suggest, merging these three states into a single one, I'll leave my reasoning in a comment below:

```suggestion
  const [product, setProduct] = useState<Product | null>(null)
```